microsoft/restler-fuzzer
# Add option to clean up resources after every sequence

In some resource-constrained environments, it is required to
clean up any resource created to test a sequence as soon as it is no longer
needed.

This change implements a new engine configuration to support this:

```run_gc_after_every_sequence```

- Also factored out garbage collector into a separate class

Testing:
- manual testing
- [TODO] add new unit tests that compare GCed objects for each
interesting test (test, test all combinations, fuzz) - they should be equal;
one new test is needed to confirm GC is being invoked when expected (perhaps
with a test hook that counts the # of invocations is equal to the # requests)
